numpyro/util.py

```diff
@@ -19,6 +21,45 @@ def set_rng_seed(rng_seed):
     onp.random.seed(rng_seed)
 
 
+def set_platform(platform=None):
+    """
+    Changes platform to CPU, GPU, or TPU. This utility only takes
+    effect at the beginning of your program.
+
+    :param str platform: either 'cpu', 'gpu', or 'tpu'.
+    """
+    if platform is None:
+        platform = os.getenv('JAX_PLATFORM_NAME', 'cpu')
+    jax.config.update('jax_platform_name', platform)
+
+
+def set_host_devices(n):
+    """
+    By default, XLA considers all CPU cores as one device. This utility tells XLA
+    that there are `n` host (CPU) devices available to use. As a consequence, this
+    allows parallel mapping in JAX :func:`jax.pmap` to work in CPU platform.
+
+    .. note:: This utility only takes effect at the beginning of your program.
+        Under the sence, this sets the environment variable
+        `XLA_FLAGS=--xla_force_host_platform_device_count=[num_devices]`, where
+        `[num_device]` is the desired number of CPU devices `n`.
+
+    .. warning:: We do not understand much the side effects when using
+        `xla_force_host_platform_device_count` flag. If you observe some strange
+        phenomenon when using this utility, please let us know through our issue
+        or forum page. Here we quote from XLA source code the meaning of this flag:
+        "Force the host platform to pretend that there are these many host
+        'devices'. All of these host devices are backed by the same threadpool.
+        Setting this to anything other than 1 can increase overhead from context
+        switching but we let the user override this behavior to help run tests
+        on the host that run models in parallel across multiple devices."
+
+    :param int n: number of CPU devices to use.
+    """
+    xla_flags = os.getenv('XLA_FLAGS', '--xla_force_host_platform_device_count={}'.format(n))
+    os.environ['XLA_FLAGS'] = xla_flags
```

Comment:
Sorry, I didn't notice this earlier - this will overwrite `XLA_FLAGS` in the environment, e.g. if there are other options to disable multithreading, fast math mode etc. We should check that and add to it instead. 

```python
xla_flags = os.getenv('XLA_FLAGS, '').split()
os.environ['XLA_FLAGS'] = ' '.join(['--xla_force_host_platform_device_count={}'.format(n)] + xla_flags)
```

Reply:
This won’t take effect if XLA flags available in the environment but your solution is better. I think we also need to check if this flag is available to override it.

Reply:
Yes, this won't actually take an effect at all. Maybe something like the following:

```python
import re 

xla_flags = os.getenv('XLA_FLAGS, '')
xla_flags = re.sub('--xla_force_host_platform_device_count=.+\s', '', xla_flags).split()
os.environ['XLA_FLAGS'] = ...
```

Reply:
Thank @neerajprad , it works like a charm! :D I just make a little adjusting regarding `--` only allowed to be at the beginning of the flags. I have checked locally that XLA accepts this pattern with multi flags.

Reply:
That looks right!